types/react/index.d.ts

```diff
@@ -310,7 +310,7 @@ declare namespace React {
 
     type SFC<P = {}> = StatelessComponent<P>;
     interface StatelessComponent<P = {}> {
-        (props: P & { children?: ReactNode }, context?: any): ReactElement<any> | null;
+        (props: P & { children?: ReactNode }, context?: any): ReactElement<any> | null | false;
```

Comment:
`true` is also an acceptable empty result (making `return !foo && ...` and `return foo || ...` actually equivalent), so this can be `| boolean`.